components/server/src/workspace/workspace-starter.ts

```diff
@@ -607,8 +607,12 @@ export class WorkspaceStarter {
             user.additionalData.ideSettings = migratted;
         }
 
-        const ideChoice = user.additionalData?.ideSettings?.defaultIde;
-        const useLatest = !!user.additionalData?.ideSettings?.useLatestVersion;
+        const preferenceIDE = user.additionalData?.ideSettings?.defaultIde;
+        const perferenceUseLatest = !!user.additionalData?.ideSettings?.useLatestVersion;
+        const ideChoice = WithEditorContext.is(workspace.context) ? workspace.context.ide : preferenceIDE;
```

Comment:
what happens if `workspace.context.ide` is invalid? Should we fall back to `preferenceIDE`?

Reply:
chooseIDE will do this step, see `workspace-starter.spec.ts`